test/functional/ui/output_spec.lua

```diff
@@ -22,6 +22,7 @@ describe("shell command :!", function()
 
   after_each(function()
     screen:detach()
+    session.command('call jobstop(b:terminal_job_id)')
```

Comment:
Interesting that the test doesn't fail even though the expected screen state (which is also asserted in before_each) isn't what we really expect.  Restoring the original assertion works, and changing __that__ assertion (e.g., dropping an 'o' in 'foo') properly fails.

Must be some waiting happening in `screen:expect` to allow the screen change to happen within some tolerance, which doesn't cause a failure here because it immediately matches.